src/iam.ts

```diff
@@ -457,20 +459,14 @@ class Iam {
       options,
     );
 
-    this.request_(
-      {
-        uri: '/iam/testPermissions',
-        qs: req,
-        useQuerystring: true,
-      },
-      (err, resp) => {
-        if (err) {
-          cb!(err, null, resp);
-          return;
-        }
-
-        const availablePermissions = Array.isArray(resp.permissions)
-          ? resp.permissions
+    void this.storageTransport
+      .makeRequest({
+        url: '/iam/testPermissions',
+        queryParameters: req as unknown as StorageQueryParameters,
+      })
+      .then(({data, resp}) => {
+        const availablePermissions = Array.isArray((data as any).permissions)
```

Comment:
If possible let's try and avoid these any casts and disabling the any linter rules.